### DISCUSSION TITLE
AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR

### DISCUSSION THREAD

**Discussion by chandar1** [aws/aws-iot-device-sdk-cpp-v2#767] [Q&A]
Hi All,

Green Grass Nucleus running in my edge device, and my application/gg component connects to the nucleus on startup fine. 

The app uses aws sdk, ipc -PublishToIotCore to publish messages to the IoT core. soon after publishing about 5-10 messages successfully, I'm getting below error, and then the application disconnects from gg nucleus  

_crt error was lib-aws-c-event-stream:  AWS_ERROR_EVENT_STREAM_RPC_PROTOCOL_ERROR failed with event stream rpc connection has encountered protocol error base status : 12 crt error 4105_

what does this mean, how could it be resolved ?

Thanks

**Comment by sfod** [comment]
We need more information to be able to debug this.
This error is usually accompanied by log messages with more details, so could you enable SDK logs and share relevant error messages here?

See https://github.com/aws/aws-iot-device-sdk-cpp-v2/blob/main/documents/FAQ.md#how-do-i-enable-logging to enable logs.
